### ISSUE TITLE
 undefined method `ulimit' for LWRP resource docker_container from cookbook docker

### ISSUE THREAD

**Issue by dfduarte** [bflad/chef-docker#335]
``` 
Good Afternoon guys,

I'm testing the Docker Cookbook here, but when I do a basic deploy (using, for example, the sample resources to deploy a busybox image, given at Chef supermarket). Chef aborts the container creation  process with the following error:

Note: I'm using the current Chef version (12.4.1), and deploying into a Ubuntu 14.04 Vagrant machine

* docker_container[an echo server] action run
    
    ================================================================================
    Error executing action `run` on resource 'docker_container[an echo server]'
    ================================================================================
    
    NoMethodError
    -------------
    undefined method `ulimit' for LWRP resource docker_container from cookbook docker
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/docker/providers/container.rb:392:in `run_cli_args'
    /var/chef/cache/cookbooks/docker/providers/container.rb:354:in `run'
    /var/chef/cache/cookbooks/docker/providers/container.rb:101:in `block in class_from_file'
    
    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/docker-init/recipes/default.rb
    
     19: docker_container 'an echo server' do
     20:   image 'busybox'
     21:   port '1234:1234'
     22:   command "nc -ll -p 1234 -e /bin/cat"
     23:   detach true
     24:   init_type false
     25: end
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/docker-init/recipes/default.rb:19:in `from_file'
    
    docker_container("an echo server") do
      action [:run]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :docker_container
      cookbook_name "docker-init"
      recipe_name "default"
      image "busybox"
      port "1234:1234"
      command "nc -ll -p 1234 -e /bin/cat"
      detach true
      cmd_timeout 60
    end
    

Running handlers:
[2015-07-08T11:46:02-07:00] ERROR: Running exception handlers
Running handlers complete
[2015-07-08T11:46:02-07:00] ERROR: Exception handlers complete
Chef Client failed. 7 resources updated in 58.222808353 seconds
[2015-07-08T11:46:04-07:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2015-07-08T11:46:04-07:00] ERROR: docker_container[an echo server] (docker-init::default line 19) had an error: NoMethodError: undefined method `ulimit' for LWRP resource docker_container from cookbook docker
[2015-07-08T11:46:04-07:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

There is some specific thing what I have to do to solve that, or I need to set "ulimit" using any additional resource, or something like that? (I've looked if the cookbook had some ulimit configuration, but there's no one).

```

**Comment by rtrauntvein** [comment]
Related to #332 maybe ?

**Comment by dfduarte** [comment]
Oh @rtrauntvein I miss these two tickets. I'm sorry... Yeay, it's related with these.

I tried the appointed solution at #331 , (add the ulimit attribute at container.rb), but it doesn't works however).

**Comment by mcoffin** [comment]
From my writeup before I realized that the issue was a duplicate.

The `docker_container` provider references the property `ulimit` on a docker_container that was removed in 28217e99f402c4fa716f2164985fc1cdf145b6aa.

Line 392 from `providers/container.rb`:
```ruby
    'ulimit' => Array(new_resource.ulimit),
```

Then this was the change in `resources/container.rb`
```diff
-attribute :ulimit, :kind_of => [String, Array]
```

**Comment by dfduarte** [comment]
Hello!

Just to notice: Remove the two attribuites (from both container's.rb), was worked and the cookbook have deployed with success.

**Comment by mcoffin** [comment]
Well, I assume they'd rather not *lose* functionality that some people might be using, so #331 is probably the preferred solution. I merged that PR in to 0.41.1 locally and it's working fine there as well.

**Comment by garthcn** [comment]
Hi guys,

Is there a recommended way of resolving this at this point? Thanks!

**Comment by dfduarte** [comment]
Well @garthcn, until @bflad corrects the problem (or anyone else), the solution appointed by @mcoffin has the best until this moment (I think).

I will try reapply the #331, because I did some issues with my Chef-server today (and I have seen this after all).

**Comment by greglanthier** [comment]
+1

**Comment by mcoffin** [comment]
@greglanthier It appears the PR was merged so this is probably closable now. Might want to update your version of the cookbook

**Comment by someara** [comment]
This is fixed up as of v0.40.3

**Comment by greglanthier** [comment]
Thanks very much @someara, @mcoffin .  I'll give 0.40.3 a spin ASAP.
